llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp

```diff
@@ -4175,6 +4175,25 @@ struct MemorySanitizerVisitor : public InstVisitor<MemorySanitizerVisitor> {
     setShadow(&I, IRB.CreateBitCast(CI, getShadowTy(&I)));
     setOriginForNaryOp(I);
   }
+  // Instrument AVX permutation intrinsic.
+  // We apply the same permutation (argument index 1) to the shadows.
+  void handleAVXVpermil2var(IntrinsicInst &I) {
+    IRBuilder<> IRB(&I);
```

Comment:
Please add some assertions that check whether the instruction is suitable for this handler:

- number of operands
- type and size of each operand
- relation of input and output operands